### PR TITLE
Minor code improvements

### PR DESCRIPTION
**Pull request by amandaJayanetti**

### CODE REVIEW

File: src/SearchWrapper.cpp [hunk 1]
```diff
@@ -139,11 +139,9 @@ Rcpp::List searchWrapper(Rcpp::CharacterVector & dbpath,Rcpp::List & queryList){
   Xapian::MSet mset = enquire.get_mset(offset, pagesize);
   
   int cols=0;
-  for (Xapian::MSetIterator m = mset.begin(); m != mset.end(); ++m) {
-    const string & data = m.get_document().get_data();
-    cols=columnNo(data);
-    break;
-  }
+  const string & data = mset.begin().get_document().get_data();
```
Comment (jaylett) [inline src/SearchWrapper.cpp]: You need to test this with a zero results MSet, where it will cause problems (because `mset.begin()` in that situation can't be dereferenced). You can compare `mset.begin()` with `mset.end()` to see if it's "real".

File: src/SearchWrapper.cpp [hunk 2]
```diff
@@ -140,7 +140,7 @@ Rcpp::List searchWrapper(Rcpp::CharacterVector & dbpath,Rcpp::List & queryList){
   
   int cols=0;
   const string & data = mset.begin().get_document().get_data();
-  cols=columnNo(data);
+  cols=lineCount(data); //data consists of row values of a number of columns in indexed data frame seperated by \n
```
Comment (jaylett) [inline src/SearchWrapper.cpp]: I'd be inclined to not have the comment here, because not everyone who needs to know is going to look here for it. It needs to go somewhere, though, maybe in the comment for `indexWrapper()` (which is what creates this structure).

**Comment by jaylett** [comment]
This looks good.

Your indentation and spacing style is pretty tight, and I think it'd be more readable if you relaxed it a little. Xapian's style is fairly readable, and we probably want it to match so that future maintenance is easier by being consistent across different parts of Xapian. I have a [draft `.clang-format` file](https://gist.github.com/jaylett/49236e4d58176a49150d794aadb769f0) that you can run over C++ source that will make it Xapian-like. (Copy/download it into your RXapian directory and use `clang-format -i -style=file <file>`.) It's not perfectly Xapian-ish, but we're hoping to gradually converge on a set of clang format options that work well.

You may want to change `ColumnLimit` to something higher (120, maybe), since your source seems to contain a fair number of long lines, but isn't particularly difficult to read because of it.

I'd do this as a separate PR. You'll also want to change your editor to use tabs where possible (8 characters), and indent 4 characters at a time.

**Comment by amandaJayanetti** [comment]
>
> I have a draft .clang-format file
> <https://gist.github.com/jaylett/49236e4d58176a49150d794aadb769f0> that
> you can run over C++ source that will make it Xapian-like. (Copy/download
> it into your RXapian directory and use clang-format -i -style=file <file>.)
>


I got the error "Can't find usable .clang-format, using LLVM style" when I
tried to use the given .clang-format file. I tried using a simpler file
with few commands and it worked fine. Could you kindly tell me what could
be causing the error?

**Comment by jaylett** [comment]
I just tried by copying the raw text out from github and it worked fine, so I suspect you either copied too much, or perhaps missed a bit off the end such as the last line.

**Comment by amandaJayanetti** [comment]
It says arguments are invalid.

YAML:4:24: error: invalid boolean
AlignAfterOpenBracket: Align

Error reading /home/amanda/Documents/R-Home/RXapian/src/.clang-format:
Invalid argument
Can't find usable .clang-format, using LLVM style

Could it be a problem with clang-format versions? I'm using clang-format-3.6

**Comment by jaylett** [comment]
Ah — yes. I'm on clang 3.9. Either cut out lines that cause problems, or (better, if you can) upgrade?

> On 29 Jun 2016, at 15:19, amandaJayanetti <notifications@github.com> wrote:
> 
> It says arguments are invalid.
> 
> YAML:4:24: error: invalid boolean
> AlignAfterOpenBracket: Align
> 
> Error reading /home/amanda/Documents/R-Home/RXapian/src/.clang-format:
> Invalid argument
> Can't find usable .clang-format, using LLVM style
> 
> Could it be a problem with clang-format versions? I'm using clang-format-3.6
> 
> On Wed, Jun 29, 2016 at 7:45 PM, James Aylett <notifications@github.com>
> wrote:
> 
> > I just tried by copying the raw text out from github and it worked fine,
> > so I suspect you either copied too much, or perhaps missed a bit off the
> > end such as the last line.
> >
> > —
> > You are receiving this because you modified the open/close state.
> > Reply to this email directly, view it on GitHub
> > <https://github.com/amandaJayanetti/RXapian/pull/8#issuecomment-229369213>,
> > or mute the thread
> > <https://github.com/notifications/unsubscribe/AH_jNqxoklOEyi1J2_DI_P3lCROO8bwPks5qQn4RgaJpZM4JAL6R>
> > .
> >
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub, or mute the thread.
> 

-- 
 James Aylett
 devfort.com — spacelog.org — tartarus.org/james/
